pkg/userdata/flatcar/provider.go

```diff
@@ -73,37 +78,39 @@ func (p Provider) UserData(req plugin.UserDataRequest) (string, error) {
 		return "", fmt.Errorf("error extracting cacert: %v", err)
 	}
 
-	// We need to reconfigure rkt to allow insecure registries in case the hyperkube image comes from an insecure registry
-	var insecureHyperkubeImage bool
-	for _, registry := range req.InsecureRegistries {
-		if strings.Contains(req.HyperkubeImage, registry) {
-			insecureHyperkubeImage = true
-		}
-	}
-
 	if flatcarConfig.DisableAutoUpdate {
 		flatcarConfig.DisableLocksmithD = true
 		flatcarConfig.DisableUpdateEngine = true
 	}
 
+	kubeletImage := poseidonImage
+	lessThen118, err := semver.NewConstraint(lessThen118Check)
+	if err != nil {
+		return "", err
+	}
+
+	if lessThen118.Check(kubeletVersion) {
+		kubeletImage = hyperkubeImage
+	}
+
 	data := struct {
 		plugin.UserDataRequest
-		ProviderSpec           *providerconfigtypes.Config
-		FlatcarConfig          *Config
-		Kubeconfig             string
-		KubernetesCACert       string
-		KubeletVersion         string
-		InsecureHyperkubeImage bool
-		NodeIPScript           string
+		ProviderSpec     *providerconfigtypes.Config
+		FlatcarConfig    *Config
+		Kubeconfig       string
+		KubernetesCACert string
+		KubeletImage     string
+		KubeletVersion   string
+		NodeIPScript     string
 	}{
-		UserDataRequest:        req,
-		ProviderSpec:           pconfig,
-		FlatcarConfig:          flatcarConfig,
-		Kubeconfig:             kubeconfigString,
-		KubernetesCACert:       kubernetesCACert,
-		KubeletVersion:         kubeletVersion.String(),
-		InsecureHyperkubeImage: insecureHyperkubeImage,
```

Comment:
This is a [machine-controller flag (`-node-hyperkube-image`)](https://github.com/kubermatic/machine-controller/blob/7e26d77173e9b4ef3b8bdf137e4d1f3a82469be2/cmd/machine-controller/main.go#L168). We should either continue using it (that would fix [this comment](https://github.com/kubermatic/machine-controller/pull/801/files#r464395765)) or we should deprecate it.

Reply:
yeah, that's the way I'm working currently on flags support